### ISSUE TITLE
Some confusion expecting a reply

### ISSUE THREAD

**Issue by Zhouwangjie** [TongfeiLiu/C3Net-for-building-extraction#1]
Great work!

I'm a beginner to deep learning. I want to make some changes on your work to improve segmentation accuracy, but something confused me when running the project. 

1. For the Inria dataset preparation, the images are 5000×5000, and for how to crop the non-overlapping clipping to 81 sheets in 512 ×512. Is the image I thought below correct？
![split](https://github.com/TongfeiLiu/C3Net-for-building-extraction/assets/62125826/fed30633-870a-4ea2-b1fb-7e98e86b466b)

2. I train 100 epoches, and save the 100th epoch model. The mIoU looks higer and I don't know the reason. (The grey is my test results.)
![image](https://github.com/TongfeiLiu/C3Net-for-building-extraction/assets/62125826/fd4a5e1c-b463-494b-8409-f59569118432)

3. I used another [dataset](https://www.mdpi.com/2072-4292/13/11/2187). It has a small number of pictures. When performing the ablation experiment, I annotated C2AM and ER2M in the network construction and only used BCE Loss, but the best mIoU result in 100 rounds of training were also achieved and even better. I don't know what's wrong here.
code:
![ablation](https://github.com/TongfeiLiu/C3Net-for-building-extraction/assets/62125826/723c250b-2efe-49c2-a026-e0435132c2e6)
C3Net-BD:
![image](https://github.com/TongfeiLiu/C3Net-for-building-extraction/assets/62125826/988de323-1bcd-45d4-aab5-92735f493a29)
C3Net-ablation-B:
![image](https://github.com/TongfeiLiu/C3Net-for-building-extraction/assets/62125826/87e2d0c2-99fc-4f39-ba44-577788463377)

4. Comparing the article and the code, this one feels inconsistent, but the experimental results feel not much different.
![image](https://github.com/TongfeiLiu/C3Net-for-building-extraction/assets/62125826/52f96e76-3471-4566-9176-fbfe7931a575)
![image](https://github.com/TongfeiLiu/C3Net-for-building-extraction/assets/62125826/31e335bc-8699-4657-bcbb-9319f417a10c)

It may be that I didn't look carefully in some places. Looking forward to your reply and hope you can solve my confusion.

**Comment by TongfeiLiu** [comment]
Thanks for your concerns and questions about our work.
1.	Data preparation is correct.
2.	This situation is very likely to occur because the performance in the paper is obtained under a fixed set of parameter settings in this paper, that is: the maximum epoch is set to 70, the learning rate is 0.0001 and the decline rate is [30, 70] per 5 down to 0.9. To sum up, different settings and epochs may obtain different results, which may be better than the paper's results, or maybe slightly lower than the accuracy of the article. Therefore, obtaining similar accuracy can meet the relevant usage requirements. If you need a model or visual results consistent with the accuracy of the paper, please contact me again.
It's worth noting that the precision you're getting (gray) seems to be wrong, it's advisable to double-check for errors.
3.	In fact, performance may vary on different datasets. This is because dice loss is often trained on a lot of data that is very unstable or even unable to converge well, which will also lead to better results using only BCE loss. So, it's not your fault.
4.	Thank you for your question, this is indeed a mistake, and corresponding changes and hints have been made in the code. We tested and verified a lot of using E5 or FCx as the input of CnAB (the original text or the scheme shown in the figure). Although the results are not much different, we think that it is more direct to use the original E5 features to enhance the details of the features after the pyramid. , which also fits our motivation. You can use a similar setup depending on the situation, we think it is a feasible solution.

**Comment by Zhouwangjie** [comment]
First of all, thank you for your sincere answer, sorry to see your reply so late. I re-did the experiments on the Austin dataset in inria, and my results still confuse me. The results without adding the two modules you mentioned are a little better than those shown in your paper, but after adding your two modules, the results are similar to yours, but the ablation results are not ideal. Not sure if I did something wrong again. It would be great if you can provide the download address of the paper ablation training model results. Looking forward to your reply, best wishes.
![image](https://github.com/TongfeiLiu/C3Net-for-building-extraction/assets/62125826/6c6ce68a-97f1-4c13-a8ab-7d8b21bf63bb)

**Comment by TongfeiLiu** [comment]
Sorry for not replying in time due to my busy graduation. Thank you so much for your attention. Only using the backbone network is the U-net network. U-net can refer to the performance in other papers. As far as I know, U-net should not achieve the performance you mentioned. So please try again if the performance of the backbone network is correct. Second, our ablation models take time to sort out, and you may have to wait a few days.

**Comment by Zhouwangjie** [comment]
> Sorry for not replying in time due to my busy graduation. Thank you so much for your attention. Only using the backbone network is the U-net network. U-net can refer to the performance in other papers. As far as I know, U-net should not achieve the performance you mentioned. So please try again if the performance of the backbone network is correct. Second, our ablation models take time to sort out, and you may have to wait a few days.

Thank you again. I re-ran the Austin experiment on pure UNet and found that the down sampling in the paper used maxpool and avepool, and the original UNet used maxpool, and I tried both. Using BCEloss alone, the iou of building can reach 81 in 80 training epoch. I validate with a test set every epoch I train and choose the best. Whether it is caused by inconsistent data set slicing processing, whether the method of data set slicing processing or the download link of the processed dataset could be disclosed. Thank you for your help sincerely. Looking forward your reply.
